src/audio_midiout.h

```diff
@@ -0,0 +1,188 @@
+/*
+ * This file is part of EasyRPG Player.
+ *
+ * EasyRPG Player is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * EasyRPG Player is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with EasyRPG Player. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef EP_AUDIO_MIDIOUT_H
+#define EP_AUDIO_MIDIOUT_H
+
+#include <string>
+#include "filesystem_stream.h"
+#include "audio_midiout_device.h"
+
+/**
+ * Manages sequencing MIDI files and emitting MIDI events to a MIDI out 
+ * device.
+ */
+class MidiOut {
+public:
+	virtual ~MidiOut() = default;
+
+	static std::unique_ptr<MidiOut> Create(Filesystem_Stream::InputStream& stream, const std::string& filename);
+
+	/**
+	 * Prepares a volume fade in/out effect.
+	 * To do a fade out begin must be larger then end.
+	 * Call Update to do the fade.
+	 * Volume changes will not really modify the volume but are only helper
+	 * functions for retrieving the volume information for the audio hardware.
+	 *
+	 * @param begin Begin volume (from 0-100)
+	 * @param end End volume (from 0-100)
+	 * @param duration Fade duration in ms
+	 */
+	virtual void SetFade(int begin, int end, int duration) {
```

Comment:
`chrono::milliseconds` for duration?

Reply:
This `SetFade` is passed the value directly from the RPG Maker game's parameters, and it's also consistent with how `AudioDecoder` works. I'm not opposed to using a better C++ type here but if we do we'd want to adopt it across the board instead of just in this one place, and I'm not sure this is the PR for that...

Reply:
Agree, let's not refactor the whole interface. This is good as it is.